### ISSUE TITLE
__resolveType and __isTypeOf do not support Promise as return value

### ISSUE THREAD

**Issue by marcoreni** [dotansimha/graphql-code-generator#3549]
**Describe the bug**
GraphQL supports returning promise for __resolveType and __isTypeOf (for interfaces and unions) since https://github.com/graphql/graphql-js/pull/631, but code generator does not support it.

1. My GraphQL schema:

```graphql
interface Item {
}
```

3. My `codegen.yml` config file:

```yml
overwrite: true
schema: "./schemas/schema.gql"
documents: null

generates:
  src/types.ts:
    plugins:
      - "typescript"
      - "typescript-resolvers"
    config:
      useIndexSignature: true
```

**Expected behavior**
Generated code for `TypeResolveFn` and `isTypeOfResolverFn` should include the Promise as a result, similar to what it's done for ResolverFn.

**Comment by dotansimha** [comment]
Working on a fix in: https://github.com/dotansimha/graphql-code-generator/pull/3599 

@marcoreni can you please try the alpha version? `1.12.3-alpha-47f7aea4.101`

**Comment by marcoreni** [comment]
I tested the alpha version and it seems to work.

Thanks, @dotansimha !

**Comment by dotansimha** [comment]
Available in `v1.13.0`
